.github/workflows/publish.yml

```diff
@@ -20,13 +24,41 @@ jobs:
           python-version: "3.x"
 
       - name: "Install dependencies"
-        run: python -m pip install build
+        run: python -m pip install build==0.8.0 sigstore==0.6.2
 
       - name: "Build dists"
         run: |
           SOURCE_DATE_EPOCH=$(git log -1 --pretty=%ct) \
           python -m build
 
+      - name: "Sign dists"
+        run: |
+          mkdir -p sigstore-artifacts/
+
+          for dist in dist/*; do
+            dist_name=$(basename "${dist}")
+
+            # Sign the dists and then verify them immediately
+            # with the generated artifacts.
+            python -m \
+              sigstore sign "${dist}" \
+              --output-signature sigstore-artifacts/"${dist_name}.sig" \
+              --output-certificate sigstore-artifacts/"${dist_name}.crt"
+
+            python -m \
+              sigstore verify "${dist}" \
+              --cert "sigstore-artifacts/${dist_name}.crt" \
+              --signature "sigstore-artifacts/${dist_name}.sig" \
+              --cert-oidc-issuer https://token.actions.githubusercontent.com
+
+          done
+
+      - uses: "actions/upload-artifact@3cea5372237819ed00197afe530f5a7ea3e805c8"
+        with:
+          name: "sigstore-artifacts"
+          path: "sigstore-artifacts/*"
+          if-no-files-found: "error"
```

Comment:
Not an issue just a thought, do we want this to be a standalone script that's invoked by the workflow? That allows this to be dry-run tested locally, doesn't couple the process to GHA.

Reply:
Unfortunately the process is coupled to GHA because we rely on GitHub's OpenID Connector being available ambiently for sigstore to sign the distributables with. I do see value in lifting this bash script into a separate file though for easier editing.